### ISSUE TITLE
When using single input, model always produces wrong output

### ISSUE THREAD

**Issue by youngkyoonjang** [mzolfaghari/ECO-efficient-video-understanding#43]
** Although I am testing PyTorch implementation, I guess it might happen to the Caffe implementation as well. So I am asking if there is an answer to my question. 

Hello,

I am Youngkyoon Jang, who is a postdoc in the University of Bristol. Thanks for sharing the PyTorch implementation of ECO network for testing new dataset, EPIC-KITCHEN. But I am facing the problem when testing the model to calculate the accuracy. 

I am currently using PyTorch implementation for ECO network. And I noticed that the model always predicts the first index with the highest score when using a single input instead of multiple mini-batch samples in testing. Did you know this problem? Is there a correct way to get a consistent output? 

When I put a different number of batch sample in a mini-batch, the model predicts a different score for the same sample depending on the number of mini-batch. 

I look forward to your reply. 

Best, 
Young

**Comment by youngkyoonjang** [comment]
I found my mistake that I did not switch to evaluation mode. 

I should have put the following:
  model.eval()
before the test loop.
